File: API/ContaCentral/ContaCentral/Domain/Models/Transferencia.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace ContaCentral.Domain.Models
{
    public class Transferencia
    {
        public int Id { get; set; }
        public decimal Valor { get; set; }
        public DateTime DataHora { get; set; }
        public string Descricao { get; set; }

        public int FromCarteiraId { get; set; }
        [ForeignKey("FromCarteiraId")]
        public Carteira FromCarteira { get; set; }

        public int ToCarteiraId { get; set; }
        [ForeignKey("ToCarteiraId")]
        public Carteira ToCarteira { get; set; }
    }
}

File: Website/src/app/domain/models/transferencia.ts
import { Carteira } from "./carteira";

export class Transferencia{
    id: number;
    valor: number;
    dataHora: Date;
    descricao: string;

    fromCarteiraId: number;
    fromCarteira: Carteira;

    toCarteiraId: number;
    toCarteira: Carteira;

    constructor(id: number, valor: number, dataHora: Date, descricao: string, fromCarteiraId: number, fromCarteira: Carteira, toCarteiraId: number, toCarteira: Carteira,){
        this.id = id;
        this.valor = valor;
        this.dataHora = dataHora;
        this.descricao = descricao;
        this.fromCarteiraId = fromCarteiraId;
        this.fromCarteira = fromCarteira;
        this.toCarteiraId = toCarteiraId;
        this.toCarteira = toCarteira;       
    }
}